--- iui.js	Mon Mar 30 18:29:07 2009
+++ iui.onload.js	Wed Apr 29 08:29:16 2009
@@ -20,6 +20,7 @@
 var hasOrientationEvent = false;
 var portraitVal = "portrait";
 var landscapeVal = "landscape";
+var loadedPages = [];
 
 // *************************************************************************************************
 
@@ -46,6 +47,14 @@
                     setTimeout(slidePages, 0, fromPage, page, backwards);
                 else
                     updatePage(page, fromPage);
+                    
+			    if (loadedPages.indexOf(page.id) == -1)
+			    {
+			    	loadedPages.push(page.id);
+   
+					if (page.getAttribute("onload"))
+						eval(page.getAttribute("onload"));
+	    		}
             }
         }
     },
