--- iui.js	Mon Mar 30 18:29:07 2009
+++ iui.onblur.js	Wed Apr 29 08:31:05 2009
@@ -43,7 +43,12 @@
                 currentPage = page;
 
                 if (fromPage)
+                {
+	                if (fromPage.onblur)
+						fromPage.onblur();
+						
                     setTimeout(slidePages, 0, fromPage, page, backwards);
+                }
                 else
                     updatePage(page, fromPage);
             }
